protocol/0019-MCAL-margin_calculator.md

```diff
@@ -84,21 +84,131 @@
+   2. If the order is not persistent this is the end, if it is persistent any portion of the order which has not traded in step 1 will move to being placed on the order book. 
+      1. At this point, the party's general account will be checked for sufficient margin to cover `new margin = limit price * remaining size * margin factor`, as this is the worst-case trade price of the remaining component. If there is insufficient, the remaining portion of the order will be `stopped`. If there is sufficient, the order margin will be calculated for the current position plus the party's orders including this potential new order. If this order margin is greater than the sum of `new margin + margin balance + order margin balance` then the order will be rejected. Otherwise, an amount `new margin` will be moved into the party's Order Margin account and the order will be placed on the book.
+
+NB: This means that a party's order could partially match, with a trade executed and some funds moved to the margin account with correct leverage whilst the rest of the order is immediately stopped.
+
+When submitting, amending, or deleting an order in isolated margin mode and an auction is active there is no concept of an order trading immediately on entry, however the case of someone putting in a sell order for a very low price must be handled (as it is likely to execute at a much higher price). To handle this, when in an auction the amount taken into the order margin account should be the larger of either `limit price * size * margin factor` or `max(mark price, indicative uncrossing price) * size * margin factor`. All other steps are as above.
+
+### Increasing Position
+
+When an order trades which increases the position (increasing the absolute value of the trader's position), the target amount to be transferred is calculated as:
+
+```math
+margin to add = margin factor * VWAP of new trades * total size of new trades
+```
+
+This will be taken by performing three steps:
+ 1. The margin to add as calculated here is compared to the margin which would have been placed into the order margin account for this order, `limit price * remaining size * margin factor`. 
+ 2. If it is equal, this amount is taken from the order margin account and placed into the margin account
+ 3. If the order traded at a price which means less margin should be taken then the amount `margin to add` above is taken from the order margin account into the margin account and the excess is returned from the order margin account to the general account
+
+NB: In implementation, for any volume that trades immediately on entry, the additional margin may be transferred directly from the general account to the margin account. 
+
+### Reducing Position
+
+When an order trades which reduces the trader's current position the amount to be withdrawn from the margin account is calculated as:
+
+```math
+margin to remove = current margin * abs(total size of new trades) / abs(position prior to trade)
+```
```

Comment:
Here's the small example I mentioned on Monday:
https://docs.google.com/spreadsheets/d/1x32LzfS6WEofnFIU-7I7oAKKz2jojUQvA0sAtqudyIo/edit?usp=sharing

Reply:
`current margin` here would be the balance in the margin account, which should be purely dependent on the price moves since average_entry_price, then you close a proportion of it based on your proportion closed. 

Checking the spreadsheet

Reply:
First thought is that there's no PnL in here, I think the first trader would have had their position closed :D.

However I'm not immediately sure that "total margin taken after a series of trades should be equal at equal VWAP" is necessarily true. It feels like removing margin based on VWAP of the trades might make it tricky mentally to think about what the remaining amount means (given that isolated margin is meant to be 'simpler', "I close half my position and half my margin is released" Is easy to reason at least. "It depends how your VWAP is affected by the closing price" less so)

Reply:
(I'm also probably being dense but I'm not sure calculating average entry price including sales like that works does it?)

Reply:
Btw. I'm on board with going for ratio of sizes as originally suggested, removed my suggestion now and closing the conversation.